src/main/webapp/app/exam/exam-scores/exam-scores-average-scores-graph.component.ts

```diff
@@ -55,146 +46,70 @@ export class ExamScoresAverageScoresGraphComponent implements OnInit, AfterViewI
     ) {}
 
     ngOnInit(): void {
-        this.averagePointsTooltip = this.translateService.instant('artemisApp.examScores.averagePointsTooltip');
         this.activatedRoute.params.subscribe((params) => {
             this.courseId = +params['courseId'];
             this.examId = +params['examId'];
         });
         this.courseService.find(this.courseId).subscribe((courseResponse) => (this.course = courseResponse.body!));
         this.initializeChart();
-        this.createCharts();
-    }
-
-    ngAfterViewInit() {
-        this.chartInstance = this.chartDirective.chart;
     }
 
     private initializeChart(): void {
-        const colors = [GraphColors.BLUE];
-        const labels = [this.averageScores.title];
-        const absoluteData = [this.averageScores.averagePoints!];
-        const relativeData: number[] = [this.averageScores.averagePercentage!];
+        this.lookup[this.averageScores.title] = { absoluteValue: this.averageScores.averagePoints! };
+        const exerciseGroupAverage = this.averageScores.averagePercentage ? this.averageScores.averagePercentage : 0;
+        this.ngxData.push({ name: this.averageScores.title, value: exerciseGroupAverage });
+        this.xScaleMax = this.xScaleMax > exerciseGroupAverage ? this.xScaleMax : exerciseGroupAverage;
         this.averageScores.exerciseResults.forEach((exercise) => {
-            labels.push(exercise.exerciseId + ' ' + exercise.title);
-            colors.push(GraphColors.DARK_BLUE);
-            absoluteData.push(exercise.averagePoints!);
-            relativeData.push(exercise.averagePercentage!);
-
-            this.height += BAR_HEIGHT;
-            this.exerciseIds.push(exercise.exerciseId);
-            this.exerciseTypes.push(exercise.exerciseType);
+            const exerciseAverage = exercise.averagePercentage ? exercise.averagePercentage : 0;
```

Comment:
[nit] I think `??` operator is more suitable here.
Same at line 67

```suggestion
            const exerciseAverage = exercise.averagePercentage ?? 0;
```

Reply:
Thanks for the hint, I was not aware of this shortcut. Done in d55f55d